### PR TITLE
Blog App - solving n+1 problems and adding integration specs

### PR DESCRIPTION
**Pull request by misalazeem**
Following requirements were implemented

n+1
Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
Do not use the bullet gem in this exercise - instead analyze the logs as showed in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
In your PR description include the screenshots of your app's logs before and after the fix

ntegration specs
Use Capybara to write integration tests for each view in your project.

Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

n+1 problems screenshots

Before: 

Screenshot 2023-08-29 023100

After:

Screenshot 2023-08-29 031222

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class PostsController < ApplicationController
   rescue_from ActiveRecord::RecordNotFound, with: :post_not_found
 
   def index
-    @posts = @user.posts
+    @posts = @user.posts.includes(:author)
   end
```
Comment (thecodechaser) [inline app/controllers/posts_controller.rb]: - [x] Please update the above code with the below to solve the N+1 problem when fetching all posts and their comments for a user.
  - `@posts = @user.posts.includes(:comments)`

File: spec/features/users_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  end
+  scenario 'visiting the user show page, you see the number of posts the user has written..' do
+    visit user_path(user)
+    expect(page).to have_content('Number of posts: 4')
+  end
+  scenario 'User Show Page, Displaying Recent 3 posts' do
+    visit user_path(user)
+    expect(page).to have_content('Tom is from Spain')
+    expect(page).not_to have_content('fist text')
+  end
```
Comment (thecodechaser) [inline spec/features/users_show_spec.rb]: - [x] Please add the below tests for the users show page as requested in the project requirements.
  - `I can see the user's first 3 posts.`
  - `I can see a button that lets me view all of a user's posts.`
  - `When I click a user's post, it redirects me to that post's show page.`
  - `When I click to see all posts, it redirects me to the user's post's index page.`

File: spec/features/posts_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    visit user_posts_path(user)
+    expect(page).to have_content('Hello World')
+    expect(page).to have_content('Hello')
+    expect(page).to have_content('first post')
+  end
+
+  scenario 'Redirecting to post page when click on the post' do
+    visit user_posts_path(user)
+    click_link 'first post'
+    expect(page).to have_current_path(user_post_path(user, post))
```
Comment (thecodechaser) [inline spec/features/posts_index_spec.rb]: - [x] Please add the below test for the posts index page as requested in the project requirements.
  - `I can see a section for pagination if there are more posts than fit on the view.`

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @misalazeem, and @cherelemma👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with integration specs. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by iambenkis — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- Has featured tests for `User` and  `Post` with show and index page :heavy_check_mark:
-  n+1 problems Solved with `includes()` :heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @iambenkis in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**